libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -363,12 +362,249 @@ void AP_MotorsMatrix::remove_motor(int8_t motor_num)
+                    add_motor(AP_MOTORS_MOT_7,  135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 4);
+                    add_motor(AP_MOTORS_MOT_8, -135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  6);
+                    success = true;
+                    break;
+                case MOTOR_FRAME_TYPE_V:
+                    add_motor(AP_MOTORS_MOT_1,   45,  0.7981f, 1);
+                    add_motor(AP_MOTORS_MOT_2,  -45, -0.7981f, 7);
+                    add_motor(AP_MOTORS_MOT_3, -135,  1.0000f, 5);
+                    add_motor(AP_MOTORS_MOT_4,  135, -1.0000f, 3);
+                    add_motor(AP_MOTORS_MOT_5,  -45,  0.7981f, 8);
+                    add_motor(AP_MOTORS_MOT_6,   45, -0.7981f, 2);
+                    add_motor(AP_MOTORS_MOT_7,  135,  1.0000f, 4);
+                    add_motor(AP_MOTORS_MOT_8, -135, -1.0000f, 6);
+                    success = true;
+                    break;
+                case MOTOR_FRAME_TYPE_H:
+                    // H frame set-up - same as X but motors spin in opposite directions
+                    add_motor(AP_MOTORS_MOT_1,   45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  1);
+                    add_motor(AP_MOTORS_MOT_2,  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 7);
+                    add_motor(AP_MOTORS_MOT_3, -135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  5);
+                    add_motor(AP_MOTORS_MOT_4,  135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 3);
+                    add_motor(AP_MOTORS_MOT_5,  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  8);
+                    add_motor(AP_MOTORS_MOT_6,   45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 2);
+                    add_motor(AP_MOTORS_MOT_7,  135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  4);
+                    add_motor(AP_MOTORS_MOT_8, -135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 6);
+                    success = true;
+                    break;
+                default:
+                    // octaquad frame class does not support this frame type
+                    break;
+            }
+            break;
+
+        case MOTOR_FRAME_Y6:
+            switch (frame_type) {
+                case MOTOR_FRAME_TYPE_NEW_PLUS:
+                    // Y6 motor definition with all top motors spinning clockwise, all bottom motors counter clockwise
+                    add_motor_raw(AP_MOTORS_MOT_1, -1.0f,  0.500f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  1);
+                    add_motor_raw(AP_MOTORS_MOT_2, -1.0f,  0.500f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 2);
+                    add_motor_raw(AP_MOTORS_MOT_3,  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  3);
+                    add_motor_raw(AP_MOTORS_MOT_4,  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 4);
+                    add_motor_raw(AP_MOTORS_MOT_5,  1.0f,  0.500f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  5);
+                    add_motor_raw(AP_MOTORS_MOT_6,  1.0f,  0.500f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 6);
+                    success = true;
+                    break;
+                default:
```

Comment:
Ok, so two separate issues here:

- the `enum motor_frame_type` has:

   ```
   MOTOR_FRAME_TYPE_Y6B = 6,
   MOTOR_FRAME_TYPE_NEW_PLUS = 10
   ```
   Which are actually the same, the new + is the Y6B. I recommend to keep Y6B with the 10 value.

- there is a value for Y6B but not a value for Y6A. Since users already know the Y6B option I would suggest we add the Y6A option. As historically the X option was used for Y6A (see MP) I think that we should support the X to also be a Y6A. All other options should be considered a bad configuration.